src/workspace.coffee

```diff
@@ -43,6 +43,12 @@ class Workspace extends Model
     @defaultDirectorySearcher = new DefaultDirectorySearcher()
     @consumeServices(@packageManager)
 
+    # One cannot simply .bind here since it could be used as a component with
+    # Etch, which means it'd be `new`d in which case `this` would the new
+    # object.
+    realThis = this
+    @buildTextEditor = (params) -> realThis.buildTextEditor_(params)
```

Comment:
We talked about this in chat, but for posterity: basically when used as `new buildTextEditor()`, `this` is always the newly allocated object, instead of what we bound:

```
$ node -e 'var A=function(){console.log(this)}; B=A.bind(1); B(); new B();'
[Number: 1]
{}